### PR TITLE
Add idUtils

### PR DESCRIPTION
**Pull request by lsenta**

### CODE REVIEW

File: src/3box.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ const PrivateStore = require('./privateStore')
 const Verified = require('./verified')
 const Space = require('./space')
 const utils = require('./utils/index')
+const { isMuportDID } = require('./utils/id')
+const idUtils = require('./utils/id')
```
Comment (oed) [inline src/3box.js]: Why import twice here?

File: src/3box.js [hunk 2]
```diff
@@ -237,7 +239,7 @@ class Box {
   }
 
   static async _getProfileOrbit (address, opts = {}) {
-    if (utils.isMuportDID(address)) {
+    if (isMuportDID(address)) {
```
Comment (oed) [inline src/3box.js]: why not `idUtils.isMuportDID`?

File: src/3box.js [hunk 3]
```diff
@@ -496,4 +498,6 @@ async function initIPFS (ipfs, iframeStore, ipfsOptions) {
   }
 }
 
+Box.idUtils = idUtils
```
Comment (oed) [inline src/3box.js]: 👍

File: src/__tests__/idUtils.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    expect(isMuportDID('abc')).toEqual(false)
+    expect(isMuportDID('did:example')).toEqual(false)
+    expect(isMuportDID('did:muport')).toEqual(false)
+    expect(isMuportDID('did:muport:Qmb9E8wLqjfAqfKhideoApU5g26Yz2Q2bSp6MSZmc5WrNr')).toEqual(true)
+  })
+
+  test('isClaim', async () => {
+    expect(await isClaim(CLAIM_1)).toEqual(true)
+    expect(await isClaim(INVALID_CLAIM_FORMAT)).toEqual(false)
+    expect(await isClaim(EXPIRED_CLAIM)).toEqual(true) // invalid claim will throw during verify
```
Comment (oed) [inline src/__tests__/idUtils.test.js]: Thanks for testing this!

File: src/utils/id.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    } catch (e) {
+      return false
+    }
+  },
+  verifyClaim: async (claim, opts = {}) => {
+    try {
+      return await didJWT.verifyJWT(claim, opts)
+    } catch (e) {
+      return null
+    }
```
Comment (oed) [inline src/utils/id.js]: Let's get rid of the try/catch and just return the result of `verifyJWT` directly!
Will be a pain to debug if invalid claims just return null regardless of the errors.

File: src/3box.js (hunk 2 above)
Comment (lsenta) [reply src/3box.js]: Fixed!

File: src/3box.js (hunk 1 above)
Comment (lsenta) [reply src/3box.js]: Fixed!

**Pull request merged**